### PR TITLE
Add Auto Reconnect setting

### LINKED ISSUE #2300: Auto-connect to server configuration
If I open VS Code with an RPG file left open (or open a local RPG file) I don't always want to connect to an IBM i.
I think auto-connect could be a really handy feature, though I think it needs a bit of configuration.

A global setting for auto-connect would be helpful (yes/no/ask me).
"Yes" would auto-connect to the last connection
"No" would never connect
"Ask Me"pop up a dialog box and ask me which system I want to connect to and give me the option to not connect to any of them.

### PR DESCRIPTION
**Pull request by sebjulliand**
### Changes
Resolves #2300 

Based on @p-behr suggestion, this PR adds a new global settings that sets the action taken out by Code for i when a member/streamfile is left open when VS Code is closed and re-opened.

image

Supported values are:
* `ask`: opens a notification asking whether or not the user wants to reconnect
image
* `always`: always reconnect
* `never`: never reconnect

If the user choose to reconnect, the member/streamfile is reloaded. Otherwise, the editor that was left open is closed.

### How to test this PR
1. Open a member/streamfile
2. Leave it open and close VS Code
3. Re-open VS Code
4. Based on the reconnect setting, code for i will reconnect, close the editor or ask the user if they want to reconnect

### Checklist
* [x] have tested my change

### CODE REVIEW

File: src/filesystems/qsys/FSUtils.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+export async function reconnectFS(uri: vscode.Uri) {
+  const reconnect = GlobalConfiguration.get<ReconnectMode>("autoReconnect") || "ask";
+  let doReconnect = false;
+  switch (reconnect) {
+    case "always":
+      doReconnect = true;
+      break;
+
+    case "ask":
+      if (await vscode.window.showInformationMessage(l10n.t("Do you want to reconnect and open {0}?", uri.path.split('/').reverse()?.[0]), l10n.t("Reconnect"))) {
```
Comment (worksofliam) [inline src/filesystems/qsys/FSUtils.ts:21]: Perhaps `path.basename(uri.path)` might be more suited?

**Review by worksofliam — CHANGES_REQUESTED** [review]
I very much like this PR, but I think the message should show what system they're reconnecting to.

> Do you want to reconnect to SYSTEM and open FILE?

File: src/filesystems/qsys/FSUtils.ts (hunk 1 above)
Comment (sebjulliand) [reply src/filesystems/qsys/FSUtils.ts:21]: Yes, that's way less clunky 😅

**Comment by sebjulliand** [comment]
There you go @worksofliam. It's all yours for a final review.
image

File: src/filesystems/qsys/FSUtils.ts (hunk 1 above)
Comment (sebjulliand) [reply src/filesystems/qsys/FSUtils.ts]: Yes, that's way less clunky 😅

**Review by worksofliam — APPROVED** [review]
Working good! I will let you merge :)

**Comment by p-behr** [comment]
This is terrific, thanks @sebjulliand

**Comment by sebjulliand** [comment]
> This is terrific, thanks @sebjulliand

My pleasure 🙏🏻

**Pull request merged**